### PR TITLE
feat: migrate babel plugin -> typescript

### LINKED ISSUE #7: babel plugin ts migration
https://github.com/crherman7/rechunk/pull/5

### PR DESCRIPTION
**Pull request by crherman7**
## Scope

The current babel plugin implementation is in javascript which works fine but does not help with current or future development or refactors since it's not typed. Migrating to typescript will help understand what potential functions or implementations are possible with babel code generation as the plugin becomes more advanced or changes.

The only impact currently will be how the plugin is referenced in a project. Instead of utilizing the `@crherman7/rechunk/plugin` path it will be `@crherman7/rechunk/dist/plugin` which is slightly uglier BUT we may be able to use package exports to have an alias to the plugin. This will only able to be tested in a non-relative usage the package. This speaks to a potential workspace refactor as well in the future so we don't have all these relative usages, babel module resolver and tsconfig references to support the relative usage the package in the example app. Things to think about.

## Implementation

Move the babel plugin into the source directory under the babel directory so it can be bundled separately outside of the cli and react-native package. Import the types for babel and make any accommodations type-wise for typescript support. Add the namespace the tsup config so that it is bundled correctly. Update the babel config in the example app as well so it's referenced correctly.

## Screenshots

<details>
  <summary>ios</summary>

  

https://github.com/crherman7/rechunk/assets/17734240/6902f2dc-ae7a-4e78-b8ce-c29bf1b7240e
</details>

<details>
  <summary>android</summary>
  
  

https://github.com/crherman7/rechunk/assets/17734240/7cb52ca0-e9ec-40ca-a0a3-6e74c8c8c741
</details>

### ios

- yarn rechunk dev-server
- yarn ios
- verify the remote component loads successfully

### android
- yarn rechunk dev-server
- adb reverse tcp:`<port>` tcp:`<port>`
- yarn android
- verify the remote component loads successfully

## Emoji Guide

**For reviewers: Emojis can be added to comments to call out blocking versus non-blocking feedback.**

E.g: Praise, minor suggestions, or clarifying questions that don’t block merging the PR.

> 🟢 Nice refactor!

> 🟡 Why was the default value removed?

E.g: Blocking feedback must be addressed before merging.

> 🔴 This change will break something important

| | | |
| --- | --- | --- |
| Blocking | 🔴 ❌ 🚨 | RED |
| Non-blocking | 🟡 💡 🤔 💭 | Yellow, thinking, etc |
| Praise | 🟢 💚 😍 👍 🙌 | Green, hearts, positive emojis, etc |

### CODE REVIEW

File: example/babel.config.js [hunk 1]
```diff
@@ -14,6 +14,6 @@ module.exports = {
         },
       },
     ],
-    [path.resolve('../plugin')],
+    [path.resolve('../dist/babel')],
```
Comment (crherman7) [inline example/babel.config.js:17]: potentially we may be able to use package exports to still reference the babel plugin in actual projects by `@crherman7/rechunk/plugin` or `@crherman7/rehcunk/babel` tbd tho.  

🟢 good thinking forward!

File: src/cli/lib/config.ts [hunk 2]
```diff
@@ -34,7 +35,7 @@ export function getBabelConfig(dir: string = process.cwd()): any {
  * @returns {object} The ReChunk configuration object.
```
Comment (devs9) [inline src/cli/lib/config.ts]: 🟡 we can more exactly specify the returning type in JSDoc

and same for `getPackageJson`

```suggestion
 * @returns {ReChunkConfig} The ReChunk configuration object.
```

**Review by devs9 — APPROVED** [review]
Great job 👍

File: src/cli/lib/config.ts (hunk 2 above)
Comment (crherman7) [reply src/cli/lib/config.ts]: great catch! updated!

**Pull request merged**